### ISSUE TITLE
Error when parsing Oracle (+) operator with ParseUnknownFunctions.IGNORE

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#10670]
Parsing this fails, when `ParseUnknownFunctions.IGNORE` is active:

```sql
select 1 from a, b where a.x(+) = b.y
```

The error is:

```
org.jooq.impl.ParserException: Identifier expected: [1:31] select 1 from a, b where a.x(+[*]) = b.y
	at org.jooq.impl.ParserContext.expected(ParserImpl.java:12109)
	at org.jooq.impl.ParserImpl.parseName(ParserImpl.java:10150)
	at org.jooq.impl.ParserImpl.parseFieldNameOrSequenceExpression(ParserImpl.java:10064)
	at org.jooq.impl.ParserImpl.parseTerm(ParserImpl.java:7224)
	at org.jooq.impl.ParserImpl.parseUnaryOps(ParserImpl.java:6485)
	at org.jooq.impl.ParserImpl.parseExp(ParserImpl.java:6460)
	at org.jooq.impl.ParserImpl.parseFactor(ParserImpl.java:6437)
	at org.jooq.impl.ParserImpl.parseSum(ParserImpl.java:6389)
	at org.jooq.impl.ParserImpl.parseNumericOp(ParserImpl.java:6374)
	at org.jooq.impl.ParserImpl.parseCollated(ParserImpl.java:6348)
	at org.jooq.impl.ParserImpl.parseConcat(ParserImpl.java:6338)
	at org.jooq.impl.ParserImpl.parsePredicate(ParserImpl.java:5267)
	at org.jooq.impl.ParserImpl.parseNot(ParserImpl.java:5197)
	at org.jooq.impl.ParserImpl.parseAnd(ParserImpl.java:5187)
	at org.jooq.impl.ParserImpl.parseOr(ParserImpl.java:5178)
	at org.jooq.impl.ParserImpl.parseField(ParserImpl.java:6255)
	at org.jooq.impl.ParserImpl.parseField(ParserImpl.java:6180)
	at org.jooq.impl.ParserImpl.parseFieldNameOrSequenceExpression(ParserImpl.java:10080)
	at org.jooq.impl.ParserImpl.parseTerm(ParserImpl.java:7224)
	at org.jooq.impl.ParserImpl.parseUnaryOps(ParserImpl.java:6483)
	at org.jooq.impl.ParserImpl.parseExp(ParserImpl.java:6460)
	at org.jooq.impl.ParserImpl.parseFactor(ParserImpl.java:6437)
	at org.jooq.impl.ParserImpl.parseSum(ParserImpl.java:6389)
	at org.jooq.impl.ParserImpl.parseNumericOp(ParserImpl.java:6374)
	at org.jooq.impl.ParserImpl.parseCollated(ParserImpl.java:6348)
	at org.jooq.impl.ParserImpl.parseConcat(ParserImpl.java:6338)
	at org.jooq.impl.ParserImpl.parsePredicate(ParserImpl.java:5267)
	at org.jooq.impl.ParserImpl.parseNot(ParserImpl.java:5197)
	at org.jooq.impl.ParserImpl.parseAnd(ParserImpl.java:5187)
	at org.jooq.impl.ParserImpl.parseOr(ParserImpl.java:5178)
	at org.jooq.impl.ParserImpl.parseCondition(ParserImpl.java:5174)
	at org.jooq.impl.ParserImpl.parseQueryPrimary(ParserImpl.java:1506)
	at org.jooq.impl.ParserImpl.parseQueryTerm(ParserImpl.java:1378)
	at org.jooq.impl.ParserImpl.parseQueryExpressionBody(ParserImpl.java:1349)
	at org.jooq.impl.ParserImpl.parseSelect(ParserImpl.java:1132)
	at org.jooq.impl.ParserImpl.parseSelect(ParserImpl.java:1127)
	at org.jooq.impl.ParserImpl.parseQuery(ParserImpl.java:984)
	at org.jooq.impl.ParserImpl.parseQuery(ParserImpl.java:668)
	at org.jooq.impl.ParserImpl.parseQuery(ParserImpl.java:662)
	at org.jooq.impl.AbstractParserTest.assertQueryEquals(AbstractParserTest.java:251)
	at org.jooq.impl.ParserImplTest.testSelectOracleOuterJoinWithParseUnknownFunctions(ParserImplTest.java:442)
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.14.0. Backport scheduled for 3.13.5 (#10673)

**Comment by lukaseder** [comment]
Related issue: #10675
